pkg/registry/core/pod/strategy_test.go

```diff
@@ -182,6 +182,31 @@ func TestMatchPod(t *testing.T) {
 			fieldSelector: fields.ParseSelectorOrDie("status.podIP=2001:db7::"),
 			expectMatch:   false,
 		},
+		{
+			in: &api.Pod{
+				Spec: api.PodSpec{
+					SecurityContext: &api.PodSecurityContext{
+						HostNetwork: true,
+					},
+				},
+			},
+			fieldSelector: fields.ParseSelectorOrDie("spec.hostNetwork=true"),
+			expectMatch:   true,
+		},
+		{
+			in: &api.Pod{
+				Spec: api.PodSpec{},
+			},
+			fieldSelector: fields.ParseSelectorOrDie("spec.hostNetwork=false"),
+			expectMatch:   true,
+		},
+		{
+			in: &api.Pod{
+				Spec: api.PodSpec{},
+			},
+			fieldSelector: fields.ParseSelectorOrDie("spec.hostNetwork=true"),
```

Comment:
Also add 1 case for matching false when explicit false and 1 when context is nil

Reply:
Added matching false when explicit false. Context is nil case already exists.

Reply:
Sorry, I meant the say a case where the field is true but the selector wants false (no match) so that all cases are covered

Reply:
done